source/extensions/common/wasm/exports.cc

```diff
@@ -701,6 +701,13 @@ Word wasi_unstable_fd_write(void* raw_context, Word fd, Word iovs, Word iovs_len
   return 0; // __WASI_ESUCCESS
 }
 
+// __wasi_errno_t __wasi_fd_read(_wasi_fd_t fd, const __wasi_iovec_t *iovs,
+//    size_t iovs_len, __wasi_size_t *nread);
+Word wasi_unstable_fd_read(void*, Word, Word, Word, Word) {
+  // Don't support reading of any files.
+  return 8; // __WASI_EBADF
```

Comment:
What about using `NOSYS` (Function not supported) or `NOTCAPABLE` (Extension: Capabilities insufficient)?

Reply:
ENOSYS works -- interestingly there's a different value in the Emscripten C code than on my Linux system but it works. ENOTCAPABLE doesn't seem to be defined at least in my Emscripten yet.

Reply:
And if that makes sense perhaps in another PR we can update some other existing APIs the same way.

Reply:
Let's go with `ENOSYS` for clearly unimplemented functions, and it definitely makes sense to update other functions as well. Thanks!

cc @jplevyak